File: automation-test-playwright/pages/auth/ForgotPasswordPage.ts
import { type Locator, type Page } from "@playwright/test";
import { BasePage } from "../core/BasePage";

export class ForgotPasswordPage extends BasePage {
  readonly emailInput: Locator;
  readonly submitButton: Locator;

  /**
   * Initializes this page object.
   */
  constructor(page: Page) {
    super(page);
    this.emailInput = this.firstVisible(
      this.anyLocator('[data-testid="forgot-password-email"]', 'input[name="email"]', 'input[type="email"]')
    );
    this.submitButton = this.firstVisible(
      this.anyLocator('[data-testid="forgot-password-submit"]', 'button[type="submit"]')
    );
  }

  /**
   * Opens the page.
   */
  async open(): Promise<void> {
    await this.visit("/forgot-password");
  }

  async waitForLoaded(): Promise<void> {
    await this.page.waitForURL(/\/forgot-password/);
    await this.waitForVisible(this.emailInput);
    await this.waitForVisible(this.submitButton);
  }

  async submitEmail(email: string): Promise<void> {
    await this.emailInput.fill(email);
    await this.submitButton.click();
  }

  async waitForPopupContains(text: string | RegExp): Promise<void> {
    await this.waitForVisible(this.toastPopup());
    await this.waitForLocatorText(this.toastPopup(), text);
  }
}

File: automation-test-selenium/src/main/java/org/suntower/pages/auth/ForgotPasswordPage.java
package org.suntower.pages.auth;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.suntower.core.BasePage;

public class ForgotPasswordPage extends BasePage {
  private final By emailInput = anyCss("[data-testid='forgot-password-email']", "input[name='email']", "input[type='email']");
  private final By submitButton = anyCss("[data-testid='forgot-password-submit']", "button[type='submit']");

  public ForgotPasswordPage(WebDriver driver) {
    super(driver);
  }

  public void open() {
    visit("/forgot-password");
  }

  public void waitForLoaded() {
    waitForUrlContains("/forgot-password");
    waitForVisible(emailInput);
    waitForVisible(submitButton);
  }

  public void submitEmail(String email) {
    fill(emailInput, email);
    click(submitButton);
  }

  public void waitForPopupContains(String regex) {
    waitForSweetAlertContainsText(regex);
  }
}
